### PR TITLE
AVRO-3961: [C] Add AVRO_INVALID to avro_type_t

### PR DESCRIPTION
**Pull request by SahilKang**
## What is the purpose of the change

`EINVAL` was already being cast to an `avro_type_t` in the public interface, and making `AVRO_INVALID` explicit in the enum removes two preexisting workarounds.

`AVRO_INVALID` uses the same value as `EINVAL` for ABI compatibility, and a test is added to ensure no collisions with the preexisting enums.

## Verifying this change

A new test is added which can be run with `make test`

## Documentation

Although `EINVAL` was already being exposed publicly as an `avro_type_t`, this introduces a new symbol so the docs and versions have been updated to reflect a change that:
* adds to the public interface
* maintains ABI compatibility

### CODE REVIEW

**Comment by SahilKang** [comment]
good catch, I updated the commit to set `AVRO_INVALID` to `EINVAL` and added a `lang/c/tests/test_avro_type_collision.c` to verify no collisions are introduced in the future

**Comment by martin-g** [comment]
@mkmkme CC

File: lang/c/tests/test_avro_type_collision.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	CASE_ASSERTION(AVRO_RECORD)
+	CASE_ASSERTION(AVRO_ENUM)
+	CASE_ASSERTION(AVRO_FIXED)
+	CASE_ASSERTION(AVRO_MAP)
+	CASE_ASSERTION(AVRO_ARRAY)
+	CASE_ASSERTION(AVRO_UNION)
+	CASE_ASSERTION(AVRO_LINK)
+	case AVRO_INVALID:
+		break;
+	}
```
[the remark this answers is not in the document]
Comment (mkmkme) [reply lang/c/tests/test_avro_type_collision.c:61]: @KalleOlaviNiemitalo I think this should be good enough for the check. If the test can't run due to the duplicate `case`, the CI run will fail and it would mean that the approach should be changed.
WDYT?

[the remark this answers is not in the document]
Comment (SahilKang) [reply lang/c/tests/test_avro_type_collision.c:61]: yeah, and I'd consider such a compilation failure a feature, analogous to a `_Static_assert`

there are other `switch` statements under `src` which will catch the scenario being tested here but I figured it's worth making the edge case explicit as a test

with that said, my goal here is to prevent the other `avro_type_t` branches from aliasing each other: do you have another preference on how to achieve that, or do you think it's better not to test for this at all

**Review by mkmkme — APPROVED** [review]
LGTM. @KalleOlaviNiemitalo, what would you say?

**Comment by mkmkme** [comment]
@KalleOlaviNiemitalo yeah, me neither, to be honest. But the change looks valid and it doesn't break any stuff in the tests. I'd vote for merging.

**Comment by martin-g** [comment]
Thank you, @SahilKang, @KalleOlaviNiemitalo & @mkmkme  !
